jdaviz/app.py

```diff
@@ -1988,6 +2002,99 @@ def _get_viewer_item(self, ref_or_id):
+
+        # remove the current selection label from the set of labels, because its ok
+        # if the new subset shares the name of the current selection (renaming to current name)
+        if subset_selected in self._reserved_labels:
+            self._reserved_labels.remove(subset_selected)
+
+        # now check `subset_name` against list of non-active current subset labels
+        # and warn and return if it is
+        if subset_name in self._reserved_labels:
+            if raise_if_invalid:
+                raise ValueError("Cannot rename subset to name of an existing subset"
+                                 f" or data item: ({subset_name}).")
+            return False
+
+        elif not subset_name.replace(" ", "").isalnum():
+            if raise_if_invalid:
+                raise ValueError("Subset labels must be purely alphanumeric")
+            return False
+
+        else:
+            split_label = subset_name.split(" ")
+            if split_label[0] == "Subset" and split_label[1].isdigit():
+                if raise_if_invalid:
+                    raise ValueError("The pattern 'Subset N' is reserved for "
+                                     "auto-generated labels")
+                return False
+
+        return True
+
+    def _rename_subset(self, old_label, new_label, subset_group=None, check_valid=True):
+        # Change the label of a subset, making sure it propagates to as many places as it can
+        # I don't think there's an easier way to get subset_group by label, it's just a tuple
+        if subset_group is None:
+            for s in self.data_collection.subset_groups:
+                if s.label == old_label:
+                    subset_group = s
+                    break
+            # If we couldn't find a matching subset group, raise an error
+            else:
+                raise ValueError(f"No subset named {old_label} to rename")
+
+        if check_valid:
+            if self._check_valid_subset_label(new_label):
+                subset_group.label = new_label
+        else:
+            subset_group.label = new_label
+
+        # Update layer icon
+        self.state.layer_icons[new_label] = self.state.layer_icons[old_label]
+        _ = self.state.layer_icons.pop(old_label)
+
+        # Updated extracted spectrum if applicable and update _update_live_plugin_results dict.
+        if self.config == 'cubeviz':
+            for d in self.data_collection:
+                if d.label.split("(")[-1].split(",")[0] == old_label:
+                    old_data_label = d.label
+                    new_data_label = d.label.replace(old_label, new_label)
+                    d.label = new_data_label
+                    self.state.layer_icons[new_data_label] = self.state.layer_icons[old_data_label]
+                    _ = self.state.layer_icons.pop(old_data_label)
+                    results_dict = d.meta['_update_live_plugin_results']
+                    results_dict['aperture'] = new_label
```

Comment:
technically this could be stored in multiple traitlets and not only aperture.  Since the information we need is stored, let's just use it:

```suggestion
                    for key in results_dict.get('_subscriptions', {}).get('subset'):
                        results_dict[key] = new_label
```

Reply:
Is it possible for something to be subscribed to multiple subsets, e.g. one spatial and one spectral? If so we shouldn't just blindly replace these entries with the new label.

Reply:
ah, yes, of course, we also need logic to check if `results_dict[key]` was originally `old_label`